From f03acc151d083a7f5eed067cf3c4453996a72d18 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:27 -0400
Subject: [tools] lib/traceevent: Add traceevent_host_bigendian function

Message-id: <1408953210-26343-80-git-send-email-jolsa@redhat.com>
Patchwork-id: 88792
O-Subject: [PATCH RHEL7.1 BZ1133083 079/322] tools lib traceevent: Add traceevent_host_bigendian function
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 91a058ad3856b0a37cb21250eb4c478d0297a1a3
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Dec 3 14:09:18 2013 +0100

description
===========
Adding traceevent_host_bigendian function to get host endianity. It's
used in following patches.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.h b/tools/lib/traceevent/event-parse.h
index a288860..54273c0 100644
--- a/tools/lib/traceevent/event-parse.h
+++ b/tools/lib/traceevent/event-parse.h
@@ -23,6 +23,7 @@
 #include <stdbool.h>
 #include <stdarg.h>
 #include <regex.h>
+#include <string.h>
 
 #ifndef __maybe_unused
 #define __maybe_unused __attribute__((unused))
@@ -527,6 +528,15 @@ __data2host8(struct pevent *pevent, unsigned long long data)
  __data2host8(pevent, __val);    \
 })
 
+static inline int traceevent_host_bigendian(void)
+{
+ unsigned char str[] = { 0x1, 0x2, 0x3, 0x4 };
+ unsigned int val;
+
+ memcpy(&val, str, 4);
+ return val == 0x01020304;
+}
+
 /* taken from kernel/trace/trace.h */
 enum trace_flag_type {
  TRACE_FLAG_IRQS_OFF  = 0x01,
-- 
1.7.1